task_sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -463,10 +467,18 @@ def final_state(self):
 
         Not valid before the process has finished.
         """
+        if self._final_state:
+            return self._final_state
         if self._exit_code == 0:
             return self._terminal_state or TerminalTIState.SUCCESS
         return TerminalTIState.FAILED
 
+    @final_state.setter
+    def final_state(self, value):
+        """Setter for final_state for certain task instance stated present in IntermediateTIState."""
+        if value not in TerminalTIState:
+            self._final_state = value
```

Comment:
Why do we need a `setter`? Shouldn't we set TI state to falil if exit_code != 0 like we already in 

https://github.com/apache/airflow/blob/4d3140dba5976acb38c3dffd4fd963514f8da634/task_sdk/src/airflow/sdk/execution_time/supervisor.py#L472-L474

Reply:
As per our discussion, keeping it for now, will remove in a follow up PR